apps/frontend/src/app/pages/profile/profile-edit/profile-edit.component.ts

```diff
@@ -119,45 +119,33 @@ export class ProfileEditComponent implements OnInit {
 
   ngOnInit(): void {
     this.route.paramMap
-      .pipe(
-        switchMap((params: ParamMap) => {
-          if (!this.localUserService.isLoggedIn) {
-            this.router.navigateByUrl('/');
-            return EMPTY;
-          }
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe((params: ParamMap) => {
+        if (!this.localUserService.isLoggedIn) {
+          this.router.navigateByUrl('/');
+          return;
+        }
 
-          if (params.has('id')) {
-            const id = Number(params.get('id'));
-            if (this.localUserService.user.value?.id !== id) {
-              this.isLocal = false;
-              return merge(
-                this.usersService
-                  .getUser(id, { expand: ['profile', 'userStats'] })
-                  .pipe(
-                    take(1),
-                    tap((user) => this.setUser(user))
-                  ),
-                this.refreshCurrentUser
-              );
-            }
+        if (params.has('id')) {
+          const id = Number(params.get('id'));
+          if (this.localUserService.user.value?.id === id) {
+            this.isLocal = true;
+          } else {
+            this.usersService
+              .getUser(id, { expand: ['profile', 'userStats'] })
+              .subscribe((user) => this.setUser(user));
           }
-
+        } else {
           this.isLocal = true;
-          this.localUserService.refreshLocalUser();
-          return this.localUserService.user.pipe(
-            tap((user) => {
-              this.isAdmin = this.localUserService.hasRole(Role.ADMIN, user);
-              this.isModerator = this.localUserService.hasRole(
-                Role.MODERATOR,
-                user
-              );
-              this.setUser(user);
-            })
-          );
-        }),
-        takeUntilDestroyed(this.destroyRef)
-      )
-      .subscribe();
+        }
+
+        this.localUserService.refreshLocalUser();
+        this.localUserService.user.subscribe((user) => {
+          this.isAdmin = this.localUserService.isAdmin;
+          this.isModOrAdmin = this.localUserService.isModOrAdmin;
+          if (this.isLocal) this.setUser(user);
+        });
+      });
```

Comment:
Getting rid of the `switchMap` here means this page may create leaky subscriptions. Usually, if you have a `.subscribe` inside of another `.subscribe`, somethings gone wrong. Angular will not necessarily destroy and recreate this component on route change, meaning the `route.paramMap` subscription could fire multiple times. Each time you're creating new subscriptions `usersService.getUser.subscribe(...)`/`this.localUserService.user.subscribe(...)`, and these aren't getting cleaned up.

`switchMap` says, when this outer observable emits, subscribe to this inner observable. When the outer observable emits *again*, it cancels the inner subscription. So we map *only the most recent* emitted values from the outer observable onto the inner observable. If a user navigates to another profile page whilst a HTTP request from `usersService.getUser` is in flight, that request just gets discard. Deleting the `switchMap` would not only keep using the request, but it may *try to update the component* with the `setUser` stuff.

Was there a bug with the page loading before? The `refreshCurrentUser` subject does look a bit weird, can't remember if I wrote that. If you need a hand with RxJS stuff I'm happy to run through it in Discord, reactive programming is by no means easy.